### ISSUE TITLE
t.i.task.Clock.advance should set the call.called attribute.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2222]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2222|
|Type|defect|
|Created|2006-11-06 17:07:53Z|

If a `DelayedCall` produced by `twisted.internet.task.Clock.callLater`
has its cancel() method called after its callback function has been called, the following exception is raised:

```
exceptions.ValueError: list.remove(x): x not in list
```

This is quite a cryptic error.

Setting call.called to 1 before calling the callback makes the cancel() call raise the proper `AlreadyCalled` exception.

<details><summary>Searchable metadata</summary>

```
trac-id__2222 2222
type__defect defect
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1162832873000000 1162832873000000
changetime__1163338643000000 1163338643000000
version__None None
owner__ 
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm set status to `closed`|
|-|-|

(In [18644]) Add timeout behaviour to xmpp iq requests and make t.i.Clock set call.called.

Author: ralphm
Reviewer: exarkun
Fixes [#2218](https://github.com/twisted/twisted/issues/2218) and [#2222](https://github.com/twisted/twisted/issues/2222)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
